nodes/generator/line_mk4.py

```diff
@@ -0,0 +1,207 @@
+    else:  # not normalized
+        if mode == "AB":
+            scale = 1 / stepsLength  # scale to AB vector size
+        elif mode == "OD":
+            scale = 1 if nn == 0 else (1 / nn)  # scale to steps size
+
+    nx, ny, nz = [nx * scale, ny * scale, nz * scale]
+
+    vec = lambda l: (v1[0] + l * nx, v1[1] + l * ny, v1[2] + l * nz)
+
+    verts = []
+    add_vert = verts.append
+    l = -stepsLength / 2 if center else 0
+    for s in [0.0] + steps:
+        l = l + s
+        add_vert(vec(l))
+    edges = [[i, i + 1] for i in range(len(steps))]
+
+    return verts, edges
+
+
+class SvSetLineDirection(bpy.types.Operator):
+    bl_label = "Set line direction"
+    bl_idname = "sv.set_line_direction"
+    bl_description = "Set the direction of the line along X, Y or Z"
+
+    direction = StringProperty(default="X")
+    caller = StringProperty("")
+
+    def execute(self, context):
+        node_tree, node_name = self.caller.split('|><|')
+        node = bpy.data.node_groups[node_tree].nodes[node_name]
+        node.direction = self.direction
+        node.mode = "OD"
+
+        return {'FINISHED'}
+
+
+class SvLineNodeMK4(bpy.types.Node, SverchCustomTreeNode):
+    ''' Line '''
+    bl_idname = 'SvLineNodeMK4'
+    bl_label = 'Line'
+    bl_icon = 'GRIP'
+
+    def update_direction(self, context):
+        self.point_V1 = [0, 0, 0]
+        self.point_V2 = directions[self.direction]
+
+    direction = StringProperty(
+        name="Direction", default="X", update=update_direction)
+
+    mode = EnumProperty(
+        name="Mode", items=modeItems,
+        default="OD", update=updateNode)
+
+    num = IntProperty(
+        name='Num Verts', description='Number of Vertices',
+        default=2, min=2, update=updateNode)
+
+    step = FloatProperty(
+        name='Step', description='Step length',
+        default=1.0, update=updateNode)
+
+    center = BoolProperty(
+        name='Center', description='Center the line',
+        default=False, update=updateNode)
+
+    normalize = BoolProperty(
+        name='Normalize', description='Normalize line to size',
+        default=False, update=updateNode)
+
+    size = FloatProperty(
+        name='Size', description='Size of line',
+        default=10.0, update=updateNode)
+
+    point_V1 = FloatVectorProperty(
+        name='V1', description='Point V1 (starting point)',
+        size=3, default=(0, 0, 0), update=updateNode)
+
+    point_V2 = FloatVectorProperty(
+        name='V2', description='Point V2 (ending point or direction)',
+        size=3, default=(1, 0, 0), update=updateNode)
+
+    def sv_init(self, context):
+        self.inputs.new('StringsSocket', "Num").prop_name = 'num'
+        self.inputs.new('StringsSocket', "Step").prop_name = 'step'
+        size_socket = self.inputs.new('StringsSocket', "Size")
+        size_socket.prop_name = 'size'
+        size_socket.hide_safe = True
+        self.inputs.new('VerticesSocket', "V1", "V1").prop_name = "point_V1"
```

Comment:
```diff
-self.inputs.new('VerticesSocket', "V1", "V1").prop_name = "point_V1"
+self.inputs.new('VerticesSocket', "V1").prop_name = "point_V1"
```
etc :) and all the other duplicates too please.  It offers generally no benefit at the moment.